src/query/expression/src/kernels/scatter.rs

```diff
@@ -41,6 +41,15 @@ use crate::Value;
 impl DataBlock {
     pub fn scatter<I>(&self, indices: &[I], scatter_size: usize) -> Result<Vec<Self>>
     where I: common_arrow::arrow::types::Index {
+        if indices.is_empty() || self.num_columns() == 0 {
+            let mut result = Vec::with_capacity(scatter_size);
+            result.push(self.clone());
+            for _ in 1..scatter_size {
+                result.push(self.slice(0..0));
```

Comment:
The num_rows of each scattered block should according to the indices.

For example: 

indices: [0, 0, 1, 2, 1], scatter: 3.

The scattered blocks' num_rows are:

[2, 2, 1]

Reply:
But the column or the slice is empty here